crates/flight_client/src/lib.rs

```diff
@@ -42,40 +42,46 @@ pub mod tls;
 
 #[derive(Debug, Snafu)]
 pub enum Error {
-    #[snafu(display("Unable to connect to server: {source}"))]
+    #[snafu(display(
+        "Unable to connect to server: {source}\nEnsure the flight endpoint is valid and reachable."
+    ))]
     UnableToConnectToServer { source: tls::Error },
 
-    #[snafu(display("Invalid metadata value: {source}"))]
+    #[snafu(display("Authentication failed: {source}\nEnsure the credentials are valid."))]
     InvalidMetadata {
         source: tonic::metadata::errors::InvalidMetadataValue,
     },
 
-    #[snafu(display("Unable to perform handshake: {source}"))]
+    #[snafu(display(
+        "Unable to perform handshake with flight endpoint: {source}\nVerify the configuration and try again."
```

Comment:
```suggestion
        "Failed to complete the handshake with the flight endpoint: {source}\nVerify authentication settings and try again."
```

Reply:
Is a handshake something a user cares about? Maybe stating the connection failed would be clearer:

```suggestion
        "Failed to connect to the Flight server: {source}\nVerify the configuration and try again."
```

Reply:
I like the more user-friendly version as well. We just need to make sure we are not using the same error message so we can distinguish between use cases where users encounter and report the error.

Reply:
I updated this error to `Failed to connect to the Flight server: Handshake failed.` To provide a user-friendly error, as well as retaining a minimal information about handshake, IMO handshake context information could be helpful for user, as well as our internal debugging purpose.